stackrox/stackrox
# chore(ui): Delete unreachable RiskAcceptance folder

### Description

### Problem

Besides that it still takes up a bundle in the build, 5 potential errors in this unreachable code blocks `patternfly/Td-dataLabel` lint rule for consistency in tables.

### Analysis

Specific application of general procedure to delete a route.

1. Find relevant feature flag in ui/apps/platform/src/routePaths.ts file:

    ```js
    featureFlagRequirements: allDisabled(['ROX_VULN_MGMT_UNIFIED_CVE_DEFERRAL']),
    ```

    Verify when feature flag was turned on in pkg/features/list.go file: 2024-06-25 in #11521

2. Besides the expected route-related files, see a bonus occurrence of feature flag and conditional rendering of elements from RiskAcceptance folder in src/Containers/VulnMgmt/Entity/Image/VulnMgmtImageOverview.js file

3. Find imports no londer used after delete RiskAcceptance folder:

    * `import { vulnerabilityStateLabels } from 'messages/vulnerability';`
    * `import { … } from 'types/vuln_request.proto'`
    * `import { addDaysToDate } from 'utils/dateUtils';`

4. Find imports used in only one container after delete RiskAcceptance folder:

    No obvious need to move the following into the one remaining container:

    * `DateTimeFormat` element by SummaryCounts
    * `useFormModal` by EmailNotifierModal

### Solution

Commit one step at a time:

1. Delete 68 files in RiskAcceptance folder by the following command in stackrox folder:

    ```sh
    git rm -rf ui/apps/platform/src/Containers/VulnMgmt/RiskAcceptance/
    ```

2. Edit route-related files.

    * src/routePaths.ts
    * src/MainPage/Body.tsx
    * src/MainPage/Sidebar/NavigationSidebar.tsx

2. Delete imports no longer used.

    * messages/vulnerability.ts file is superseded by VulnerabilityStateTabs.tsx file.
    * types/vuln_request.proto.ts file is superseded by VulnerabilityExceptionService.ts file.
    * `addDaysToDate` in utils/dateUtils.ts file is superseded by `ExpiryField` element and `formValuesToDeferralRequest` function.

4. Delete conditional rendering in `VulnMgmtImageOverview` element.

### User-facing documentation


### Testing and quality


#### Automated testing


#### How I validated my change

1. `npm run lint` in ui/apps/platform
2. `npm run build` in ui/apps/platform and then compute branch - master for the following
    * `wc build/static/js/*.js`
        main.js -1001 = 4618974 - 4619995
        total -163709 = 11581593 - 11745302
    * `ls -al build/static/js/*.js | wc`
        files -1 = 179 - 180
3. `npm run start` in ui/apps/platform

#### Manual testing

1. Visit /main/vulnerability-management/risk-acceptance

    See **Cannot find the page**
    ![route](https://github.com/user-attachments/assets/b3dfbbc0-03ca-464c-96b3-fbe6850ef487)

    There is a subtle difference for title in browser tab and history.
    Before changes, because route exists: StackRox | Risk Acceptance
    After changes, because route does not exist: StackRox

2. Visit /main/vulnerability-management/images and then click an image link.

    See conditional rendering of link instead of CVEs under **Image Findings**
    ![image](https://github.com/user-attachments/assets/2c104256-bf5a-401e-953a-da773d173441)
